src/invidious/videos/parser.cr

```diff
@@ -132,10 +132,33 @@ def extract_video_info(video_id : String)
     params.delete("reason")
   end
 
-  {"captions", "playabilityStatus", "playerConfig", "storyboards", "streamingData"}.each do |f|
+  {"captions", "playabilityStatus", "playerConfig", "storyboards"}.each do |f|
     params[f] = player_response[f] if player_response[f]?
   end
 
+  # This part is complex just because we can't directly edit inside a JSON::Any
+  # object, so we have to unpack everything, just to repack it afterwards.
+  if streaming_data = player_response["streamingData"]?.try &.as_h
+    new_streaming_data = {} of String => JSON::Any
+
+    streaming_data.each do |key, value|
+      if key == "formats" || key == "adaptiveFormats"
+        new_formats = [] of JSON::Any
+
+        value.as_a.map(&.as_h).each do |fmt|
+          fmt["url"] = JSON::Any.new(convert_url(fmt))
+          new_formats << JSON::Any.new(fmt)
+        end
+
+        new_streaming_data[key] = JSON::Any.new(new_formats)
+      else
+        new_streaming_data[key] = value
+      end
+    end
+
+    params["streamingData"] = JSON::Any.new(new_streaming_data)
+  end
+
```

Comment:
**suggestion:** It's entirely possible to mutate data in a `JSON::Any`. So I think this can be simplified quite a lot to something like this:
```cr
  if streaming_data = player_response["streamingData"]?
    %w[formats adaptiveFormats].each do |key|
      streaming_data.as_h[key]?.try &.as_a.each do |format|
        format.as_h["url"] = JSON::Any.new(convert_url(format))
      end
    end
  end
```

Reply:
Thanks a lot! Now that seems obvious ^^